build/scripts/package.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-nocheck <-- This is needed as this file is basically a JavaScript script
+//                 but with some TypeScript niceness baked in
+const fs = require('fs')
+const path = require('path')
+const glob = require('glob')
+const execSync = require('child_process').execSync
+const pkg_dir = 'dist/package'
+
+async function copyGlob(pattern, dir = '.') {
+  glob(pattern, { cwd: dir }, (error, files) => {
+    for (var i = 0; i < files.length; i++) {
+      let src = path.join(dir, files[i])
+      let dst = path.join(pkg_dir, files[i])
+      let dstDir = path.dirname(dst)
+
+      fs.mkdirSync(dstDir, { recursive: true })
+
+      if (fs.statSync(src).isFile()) {
+        fs.copyFileSync(src, dst)
+      }
```

Comment:
This means we ignore directories? As long as are globs only care about files in .vscodeignore I guess that's fine.

I'm not sure what the right behavior should be if a glob did match a directory. Would it mean we would want to recursively copy the directory? Or maybe ignoring is the right behavior. I don't feel strongly about it, so this behavior is fine and we can deal with it when it comes up.

Reply:
So there would be two options:

- List over sub directories inside of `.vscodeignore` and then skip over them in the glob since the allowed ones will be listed in the file (currently being done)
- Only list main directory and recursively call glob instead of listing sub dirs in the `.vscodeignore`

Reply:
I say leave it as is. We can decide how to handle this if it ever comes up.

Reply:
Alright makes sense